File: fall2019/3302/HMWK_02_cfj2309/hmwk_02.java
// Jakins, Christopher
// cfj2309
// 2019-09-28

import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Arrays;
import java.util.regex.Pattern;

public class hmwk_02 {
  final static private Pattern effpea = Pattern.compile("[0-9a-fA-F]+\\(}[aeiouAEIOU]+");
  final static private Pattern stir = Pattern.compile("\\)[A-Mn-z]*\\{");
  final static private Pattern ent = Pattern.compile("#@+[\\dR-W]*\\.");

  public static void processToken( String token )
  {
      String result = "does not match.";
      if (Pattern.matches(effpea.toString(), token)) {
          result = "matches EffPea.";
      }
      else if (Pattern.matches(stir.toString(), token)) {
          result = "matches Stir.";
      }
      else if (Pattern.matches(ent.toString(), token)) {
          result = "matches Ent.";
      }
      System.out.println( ">" + token + "< " + result);
  }

  public static void main( String[] args )
  {
    System.out.println( "processing tokens from " + args[ 0 ] + " ..." );

    try {
      Files.lines( Paths.get( args[ 0 ] ) ).filter(line -> line.length() > 0).forEachOrdered(
        line -> Arrays.stream( line.split( "\\s+" )  )
            .forEachOrdered( token -> processToken( token ) ) );
    } catch ( java.io.IOException e ) {
      e.printStackTrace();
    }
  }
}

File: fall2019/3302/HMWK_02_cfj2309/hmwk_02.cs
// Jakins, Christopher
// cfj2309
// 2019-09-28

using System;
using System.IO;
using System.Text.RegularExpressions;

public class hmwk_02
{
  private static Regex effpea = new Regex(@"^[0-9a-fA-F]+\(}[aeiouAEIOU]+$");
  private static Regex stir = new Regex(@"^\)[A-Mn-z]*\{$");
  private static Regex ent = new Regex(@"^#@+[\dR-W]*\.$");

  static public void processToken( string token ) {
      string result = "does not match.";

      if (Regex.Match(token, effpea.ToString()).Success) {
          result = "matches EffPea.";
      }
      else if (Regex.Match(token, stir.ToString()).Success) {
          result = "matches Stir.";
      }
      else if (Regex.Match(token, ent.ToString()).Success) {
          result = "matches Ent.";
      }

      Console.WriteLine( ">" + token + "< " + result);
  }

  static public void Main( string[] args )
  {
    Console.WriteLine( "processing tokens from " + args[ 0 ] + " ..." );

    foreach ( string line in File.ReadAllLines( args[ 0 ] ) ) {
      foreach ( string token in line.Split( (char []) null, StringSplitOptions.RemoveEmptyEntries ) ) {
        processToken( token );
      }
    }
  }
}
